### PR TITLE
Restore solidus/tax_category_deprecated warning cop

### PR DESCRIPTION
**Pull request by safafa**
**Description**

This PR restores solidus/tax_category cop 
issue [53](https://github.com/solidusio/rubocop-solidus/issues/53)

-----------------

**Severity:**

* [ ] - info
* [ ] - refactor
* [ ] - convention (default)
* [x] - warning
* [ ] - error
* [ ] - fatal

-----------------

**Wrong Code**

```rb
  model.tax_category = data
```

**Correct Code**

```rb
  model.tax_categories = [data]
```

-----------------

**Solidus PR Link:** # Replace with link to Solidus PR where the change has been made.

-----------------

**Before submitting the PR make sure the following are checked:**

* [x] The PR relates to *only* one cop with a clear title and description.
* [x] Feature branch is up-to-date with `main` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran and ensured all tests are passing on a development environment.
* [x] If this is a new cop, added an entry for the cop on `/config/default.yml`
* [x] Updated Changelog

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -73,3 +73,8 @@ Solidus/SpreeTDeprecated:
   Enabled: true
   VersionAdded: '0.1'
   Reference: 'https://github.com/solidusio/rubocop-solidus/issues/22'
+
+Solidus/TaxCategoryDeprecated:
+  Description: 'Checks if tax_category is being used and issue a warning'
+  Enabled: true
+  VersionAdded: '<<next>>'
```
Comment (MassimilianoLattanzio) [inline config/default.yml]: Can you add a 'Reference` line linking the issue number?

Something like

```
Reference: 'https://github.com/solidusio/rubocop-solidus/issues/60'
```

File: spec/rubocop/cop/solidus/tax_category_deprecated_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Solidus::TaxCategoryDeprecated, :config do
+  let(:config) { RuboCop::Config.new }
+
+  # TODO: Write test code
```
Comment (MassimilianoLattanzio) [inline spec/rubocop/cop/solidus/tax_category_deprecated_spec.rb:6]: Nitpick: you can remove these comments

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 ## main (unreleased)
 
+### New features
+
+* [#x](https://github.com/solidusio/rubocop-solidus/pull/x): Add solidus/tax_category_deprecated warning cop. ([@safafa][])
```
Comment (MassimilianoLattanzio) [inline CHANGELOG.md:7]: Same here about the PR/issue number

File: relnotes/v0.3.0.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+### New features
+
+* [#x](https://github.com/solidusio/rubocop-solidus/pull/x): Add solidus/tax_category_deprecated warning cop. ([@safafa][])
```
Comment (MassimilianoLattanzio) [inline relnotes/v0.3.0.md:3]: Same here about the PR/issue number

**Review by MassimilianoLattanzio — APPROVED** [review]
Great job! Everything looks good.

I just left some comments regarding some nitpicks.

**Comment by safafa** [comment]
Thank you @MassimilianoLattanzio I made all the requested changes [here](https://github.com/solidusio/rubocop-solidus/compare/20dfd00268fff3afd77a34070b2e1821f8da1bb0..cd5a56e3e90f560c48b2a0eb7cca9e5311568795)

**Review by DanielePalombo — APPROVED** [review]

**Review by DanielePalombo — APPROVED** [review]

**Pull request merged**